lib/utils.js

```diff
@@ -41,18 +39,25 @@ function parseFeedContent(feeds) {
 }
 
 async function bundleFeeds(feeds, type, options) {
+    const bundleWorker = new Worker(require.resolve('./bundler'));
```

Comment:
this is super wasteful as it'll spin up new processes all the time instead of reusing them between invocation.

What are you trying to solve?

Might be a good idea to wrap it in a class or something if your issue is cleaning it up in tests. Then it's easier to set up and tear down.

Reply:
Basically, I was attempting to clean up the code there a bit and yes make testing a bit nicer. Having a module global there doesn't sit well with me. Sure we could wrap it in a class but we'd only partially solve the issue since an external party needs to worry about teardown.

Also, performance here seems a bit dependant on what you are looking at:
ie.
* its orders of magnitude worse to spin up and tear down each time if you are only measuring the time the workers are taking to spin up/close down and not the bundling time.
* in terms of actual time taken, however, setup and teardown only cost you 220 ms whereas bundling is probably anywhere from a few seconds to a minute.

Is there anything I'm missing about workers that could degrade things further? Like if the server was hit with shit loads of bundling requests all at the same time what would spinning up multiple workers do? Could you overload the system with worker spinups?

Reply:
Looks like it uses childprocess.fork under the hood which would be problematic under heavy parallelisation I would think. Looks like the class approach is probably worth the trouble then alas.

Reply:
>Sure we could wrap it in a class but we'd only partially solve the issue since an external party needs to worry about teardown.

Third party has to care about teardown anyways - this is a web server. Sounds more like a documentation issue.

>Like if the server was hit with shit loads of bundling requests all at the same time what would spinning up multiple workers do?

Potentially buckle under cpu starvation as there no longer is a managed queue of work to do, you'll just spin up new processes until the server goes down

Reply:
That said, I haven't measured it